zerver/actions/streams.py

```diff
@@ -68,6 +71,21 @@
 def do_deactivate_stream(
     stream: Stream, log: bool = True, *, acting_user: Optional[UserProfile]
 ) -> None:
+    # fixes a bug where realm settings are cached in the stream object, but in general it's
+    # a good idea to flush cache and reload before a destructive action like this.
+    flush_realm(instance=stream.realm)
+    flush_stream(instance=stream)
+
+    # Error if this stream is used for reporting messages. Note that this also works for
+    # merge_streams admin command on the command-line.
+    reportmsg_stream = stream.realm.get_reportmsg_stream()
+    if reportmsg_stream is not None and reportmsg_stream.id == stream.id:
+        raise JsonableError(
+            _(
+                "Error: cannot delete stream because it is being used to receive reporting messages. Please change this first, in the admin settings."
```

Comment:
Stylistically, I think our error messages are usually styled more like "Cannot deactivate abuse report stream."

We don't want "Error:" here -- the client would add that if appropriate in how it wants to display something.

Reply:
DONE - sorry! I'm still learning.